File: src/Domain/Library/Book.cs
using SharedKernel;

namespace Domain.Library;

public sealed class Book : Entity
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Translator { get; set; } = string.Empty;
    public string Editor { get; set; } = string.Empty;
    public string Corrector { get; set; } = string.Empty;
    public int PublisherId { get; set; }
    public int PublishPlaceId { get; set; }
    public string PublishYear { get; set; } = string.Empty;
    public string PublishOrder { get; set; } = string.Empty;
    public string Isbn { get; set; } = string.Empty;
    public string No { get; set; } = string.Empty;
    public int VolumeCount { get; set; }
    public int LanguageId { get; set; }
    public int SubjectId { get; set; }
    public string BookShelfRow { get; set; } = string.Empty;

    public Language? Language { get; set; }
    public Subject? Subject { get; set; }
    public City? City { get; set; }
    public Publisher? Publisher { get; set; }
}

File: src/Web.UI/src/app/models/book.ts
export type Book = {
  id: number;
  name: string;
  author: string;
  translator: string;
  editor: string;
  corrector: string;
  publisherId: number;
  publisherName?: string;
  publishPlaceId: number;
  publishPlaceName?: string;
  publishYear: string;
  publishOrder: string;
  isbn: string;
  no: string;
  volumeCount: number;
  languageId: number;
  languageName?: string;
  subjectId: number;
  subjectTitle?: string;
  bookShelfRow: string;
};

export type BookList = {
  id: number;
  name: string;
  author: string;
  publisherName?: string;
  translator: string;
  corrector: string;
  no: string;
  isbn: string;
  volumeCount: number;
  bookShelfRow: string;
};

export type CreateBookRequest = {
  name: string;
  author: string;
  translator: string;
  editor: string;
  corrector: string;
  publisherId: number;
  publishPlaceId: number;
  publishYear: string;
  publishOrder: string;
  isbn: string;
  no: string;
  volumeCount: number;
  languageId: number;
  subjectId: number;
  bookShelfRow: string;
};

export type UpdateBookRequest = CreateBookRequest & {
  id: number;
};

export type BookSearchRequest = {
  name?: string;
  author?: string;
  publisherId?: number;
  publishPlaceId?: number;
  languageId?: number;
  subjectId?: number;
  publishYear?: string;
};

export type BookSearchResponse = {
  id: number;
  name: string;
  author: string;
  publisherName?: string;
  publishPlaceName?: string;
  publishYear: string;
  languageName?: string;
  subjectTitle?: string;
  volumeCount: number;
  bookShelfRow: string;
};


